x/gamm/pool-models/stableswap/amm.go

```
@@ -148,6 +149,39 @@ func solveCfmm(xReserve, yReserve, yIn sdk.Dec) sdk.Dec {
 	return a
 }
 
+func approx_eq(a, b, tol sdk.Dec) bool {
+	diff := a.Sub(b).Abs()
+	return diff.Quo(a).LTE(tol) && diff.Quo(b).LTE(tol)
+}
+
+var (
+	twodec, _    = sdk.NewDecFromStr("2.0")
```

use `MustNewDecFromStr` to avoid ignoring the error.